pontoon/base/views.py

```diff
@@ -304,9 +305,23 @@ def get_translations_from_other_locales(request):
         approved=True
     ).order_by('locale__name')
 
-    payload = list(translations.values(
+    other_locales = list(translations.values(
         'locale__code', 'locale__name', 'locale__direction', 'locale__script', 'string'
     ))
+
+    preferred_source_translations = Translation.objects.filter(
+        entity=entity,
+        locale__in=selected_locales,
+        plural_form=plural_form,
+        approved=True
+    )
+
+    preferred_locales = list(preferred_source_translations.values(
+        'locale__code', 'locale__name', 'locale__direction', 'locale__script', 'string'
+    ))
```

Comment:
I would create a helper function that would return a list of requested values for both, preferred and other locales. That way you only need to list values once.